### PR TITLE
Tests to check lvalue expressions in depend clause.

### PR DESCRIPTION
**Pull request by rjenaa**

### CODE REVIEW

File: tests/5.0/task/test_task_depend_lvalue_func.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+int var;
+
+// Function to return reference to var
+int& ref() {
+	return var;
+}
+
+int test_task_depend_lvalue_func() {
+	int errors = 0;
+	int value;
```
Comment (seyonglee) [inline tests/5.0/task/test_task_depend_lvalue_func.cpp:25]: Explicitly initialize `value` here.

File: tests/5.0/task/test_task_depend_lvalue_ptr.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+#include <omp.h>
+#include <iostream>
+#include <cstdlib>
+#include "ompvv.h"
+
+int test_task_depend_lvalue_ptr() {
+	int errors = 0;
+	int *ptr = new int;
+	int value;
```
Comment (seyonglee) [inline tests/5.0/task/test_task_depend_lvalue_ptr.cpp:18]: Explicitly initialize `value` here.

File: tests/5.0/task/test_task_depend_lvalue_ternary.cpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+#include <iostream>
+#include <cstdlib>
+#include "ompvv.h"
+
+int test_task_depend_ternary() {
+	int errors = 0;
+	int a = 1;
+	int b = 0;
+	int c = -1;
+	int value;
```
Comment (seyonglee) [inline tests/5.0/task/test_task_depend_lvalue_ternary.cpp:20]: Explicitly initialize `value` here too.

**Comment by spophale** [comment]
@rjenaa will convert these to offloading tests.

File: tests/5.0/target/test_target_depend_lvalue_func.cpp [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+// Function to return reference to var
+int& ref() {
+	return var;
+}
+#pragma omp end declare target
+int test_task_depend_lvalue_func() {
+	int errors = 0;
+	int value = 0;
+	var = 0;
+	#pragma target omp parallel map(tofrom: value, ref())
```
Comment (spophale) [inline tests/5.0/target/test_target_depend_lvalue_func.cpp:28]: remove ref() from mapping.

**Comment by spophale** [comment]
GCC 13.2.1 output for function test:

error: expected identifier or '(' before '&' token
   19 | int& ref() {
      |    ^
test_ray3.c: In function 'test_task_depend_lvalue_func':
test_ray3.c:36:54: error: 'ref' undeclared (first use in this function)
   36 |                         #pragma omp task depend(out: ref())
      |                                                      ^~~
test_ray3.c:36:54: note: each undeclared identifier is reported only once for each function it appears in
test_ray3.c:38:33: warning: implicit declaration of function 'ref' [-Wimplicit-function-declaration]
   38 |                                 ref() = 1;
      |                                 ^~~
test_ray3.c:38:39: error: lvalue required as left operand of assignment
   38 |                                 ref() = 1;
      |                                       ^

**Review by spophale — APPROVED** [review]

**Review by andrewkallai — APPROVED** [review]

**Pull request merged**